C#/Botkit/Microsoft.BotKit.Adapters.Slack/SlackBotWorker.cs

```diff
@@ -29,7 +51,31 @@ public SlackBotWorker(Botkit botkit, object config) : base(botkit, config)
         /// <param name="userId">A Slack user id, like one found in `message.user` or in a `<@mention>`</param>
         public async Task<object> StartPrivateConversation(string userId)
         {
-            return new object();
+            try
+            {
+                JoinDirectMessageChannelResponse actual = null;
+
+                // TO-DO: await this.api.im.open({ user: userId });
+                this.api.JoinDirectMessageChannel(response =>
+                {
+                    actual = response;
+                }, userId);
+
+                var convRef = new ConversationReference();
+                var activity = (Activity)this.GetConfig("activity");
+
+                convRef.Conversation.Id = actual.channel.id;
+                (convRef.Conversation as dynamic).Team = (activity.Conversation as dynamic).Team;
+                convRef.User.Id = userId;
+                convRef.User.Name = null;
+                convRef.ChannelId = "slack";
+
+                return await ChangeContext(convRef);
+            }
+            catch(Exception ex)
```

Comment:
As in this `Catch` throwing the same exception, I propose to remove the _try/catch_ blocks. These blocks should be used when this method is called outside the package.

Reply:
Done. Thanks!